cpp/build-support/run_cpplint.py

```diff
@@ -0,0 +1,108 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import argparse
+import multiprocessing as mp
+import sys
+import lintutils
+from functools import partial
+import json
+
+_filters = '''
+-whitespace/comments
+-readability/todo
+-build/header_guard
+-build/c++11
+-runtime/references
+-build/include_order
+'''.split()
+
+_exclude_globs = '''
+*_generated.h
+*pyarrow_api.h
+*pyarrow_lib.h
+*config.h
+*vendored*
+*zmalloc.h
+*ae.h
+'''.split()
```

Comment:
It would be a bit more maintainable to keep the exclusions in a file

Reply:
Maintainable maybe wrong word -- more _consistent_ with some other places where exclusions are tracked.

Reply:
The exclusions for clang-format in `clang_format_exclusions.txt` look identical to these, should I just rename that file and use it for all three?